aggregators/rules/ipg.js

```diff
@@ -17,35 +27,65 @@ class IPG {
 
                     const article =  $("article");
                     article.find(".entry-header, .page-navigation, em").remove();
 
                     const that = this;
-                    article.find("h2, p, li, td, .card-header").each(function(){
-                        let text = $(this).text()+"\n";
-                        const nodeName = $(this).prop("nodeName").toLowerCase();
-                        if(nodeName==="h2" || nodeName==="div"){
-                            text = "\n**"+text+"**";
-                        }
-                        if(result.join("").length+text.length>that.maxLength){
-                            callback(result.join("").replace(/\n\s*\n\s*\n/g, '\n\n'));
-                            result = [];
-                        }
-                        result.push(text);
-                    });
-                    callback(result.join("").replace(/\n\s*\n\s*\n/g, '\n\n'));
+                    if(parameters.length === 1){
+                        article.find("td").each(function(){
+                            result.push($(this).text()+"\n");
+                        });
+                        article.find("h2, p, li, .card-header").each(function(){
+                            let text = $(this).text()+"\n";
+                            const nodeName = $(this).prop("nodeName").toLowerCase();
+                            if(nodeName==="h2" || nodeName==="div"){
+                                text = "\n**"+text+"**";
+                            }
+                            result.push(text);
+                            if(result.join("").length>that.maxPreview){
+                                return false;
+                            }
+                        });
+                        callback(result.join("").replace(/\n\s*\n\s*\n/g, '\n\n')+"<"+url+">");
+                    }else{
+                        const topics = parameters.slice(1,parameters.length);
+                        let writeToggle = false;
+                        article.find("h2, p, li, .card-header").each(function(){
+                            let text = $(this).text();
+                            const nodeName = $(this).prop("nodeName").toLowerCase();
+                            if(nodeName==="h2" || nodeName==="div"){
+                                if(topics.includes(handlers[text.trim()])) {
+                                    writeToggle = true;
+                                    text = "\n**"+text+"\n**";
+                                }else{
+                                    writeToggle = false;
+                                    return;
+                                }
+                            }else{
+                                text = text +"\n";
+                            }
+                            if(!writeToggle){
+                                return;
+                            }
+                            if(result.join("").length+text.length>that.maxLength){
```

Comment:
We don't need to do the splitting ourselves. discord.js has support for that, see https://discord.js.org/#!/docs/tag/master/typedef/MessageOptions .